feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go

```diff
@@ -0,0 +1,321 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core_lldp_tlv_population_test
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/open-traffic-generator/snappi/gosnappi"
+	"github.com/openconfig/featureprofiles/internal/attrs"
+	"github.com/openconfig/featureprofiles/internal/confirm"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/featureprofiles/internal/otgutils"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	otgtelemetry "github.com/openconfig/ondatra/gnmi/otg"
+	"github.com/openconfig/ondatra/otg"
+	"github.com/openconfig/ygnmi/ygnmi"
+)
+
+type lldpTestParameters struct {
+	systemName string
+	macAddress string
+	otgName    string
+}
+
+type lldpNeighbors struct {
+	systemName    string
+	portId        string
+	portIdType    otgtelemetry.E_LldpNeighbor_PortIdType
+	chassisId     string
+	chassisIdType otgtelemetry.E_LldpNeighbor_ChassisIdType
+}
+
+const (
+	portName = "port1"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+var (
+	// Ethernet configuration.
+	ateSrc = attrs.Attributes{
+		Name: "ateSrc",
+		MAC:  "02:00:01:01:01:01",
+	}
+
+	// LLDP configuration.
+	lldpSrc = lldpTestParameters{
+		systemName: "ixia-otg",
+		macAddress: "02:00:22:01:01:01",
+		otgName:    "ixia-otg",
+	}
+)
+
+// Determine LLDP advertisement and reception operates correctly.
+func TestCoreLLDPTLVPopulation(t *testing.T) {
+	tests := []struct {
+		desc        string
+		lldpEnabled bool
+	}{
+		{
+			desc:        "LLDP Enabled",
+			lldpEnabled: true,
+		},
+		{
+			desc:        "LLDP Disabled",
+			lldpEnabled: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+
+			// DUT configuration.
+			t.Log("Configure DUT.")
+			dut, dutConf := configureDUT(t, "dut", test.lldpEnabled)
+			dutPort := dut.Port(t, portName)
+			verifyNodeConfig(t, dut, dutPort, dutConf, test.lldpEnabled)
+
+			// ATE Configuration.
+			t.Log("Configure ATE")
+			ate := ondatra.ATE(t, "ate")
+			otg := ate.OTG()
+			otgConfig := configureATE(t, otg)
+
+			checkOtgLLDPMetrics(t, otg, otgConfig, test.lldpEnabled)
+
+			dutPeerState := lldpNeighbors{
+				systemName:    lldpSrc.systemName,
+				chassisId:     lldpSrc.macAddress,
+				chassisIdType: otgtelemetry.LldpNeighbor_ChassisIdType_MAC_ADDRESS,
+				portId:        ate.Port(t, portName).Name(),
+				portIdType:    otgtelemetry.LldpNeighbor_PortIdType_INTERFACE_NAME,
+			}
+
+			if test.lldpEnabled {
+				expOtgLLDPNeighbors := map[string][]lldpNeighbors{
```

Comment:
Rather than test.lldpEnabled, make the wantLLDPNeighborsOTG part of the test struct? Also, split verifyDUTTelemetry into two functions: verifyDUTEnableLLDP and verifyDUTDisableLLDP and pass them as test.verifyDUT.

The main idea is to avoid having lengthy if-then-else (or switch statements) in the for loop of the test cases.

See: https://github.com/golang/go/wiki/TestComments#table-driven-tests-vs-multiple-test-functions

Reply:
verifyDUTTelemetry is only used when lldpEnabled is true so we don't need to split.